tests/components/network/test_init.py

```diff
@@ -0,0 +1,446 @@
+"""Test the Network Configuration."""
+from unittest.mock import Mock, patch
+
+import ifaddr
+
+from homeassistant.components import network
+from homeassistant.components.network.const import (
+    ATTR_ADAPTERS,
+    ATTR_CONFIGURED_ADAPTERS,
+    STORAGE_KEY,
+    STORAGE_VERSION,
+)
+from homeassistant.setup import async_setup_component
+
+_NO_LOOPBACK_IPADDR = "192.168.1.5"
+_LOOPBACK_IPADDR = "127.0.0.1"
+
+
+def _generate_mock_adapters():
+    mock_lo0 = Mock(spec=ifaddr.Adapter)
+    mock_lo0.nice_name = "lo0"
+    mock_lo0.ips = [ifaddr.IP("127.0.0.1", 8, "lo0")]
+    mock_eth0 = Mock(spec=ifaddr.Adapter)
+    mock_eth0.nice_name = "eth0"
+    mock_eth0.ips = [ifaddr.IP(("2001:db8::", 1, 1), 8, "eth0")]
+    mock_eth1 = Mock(spec=ifaddr.Adapter)
+    mock_eth1.nice_name = "eth1"
+    mock_eth1.ips = [ifaddr.IP("192.168.1.5", 23, "eth1")]
+    mock_vtun0 = Mock(spec=ifaddr.Adapter)
+    mock_vtun0.nice_name = "vtun0"
+    mock_vtun0.ips = [ifaddr.IP("169.254.3.2", 16, "vtun0")]
+    return [mock_eth0, mock_lo0, mock_eth1, mock_vtun0]
+
+
+async def test_async_detect_interfaces_setting_non_loopback_route(hass, hass_storage):
+    """Test without default interface config and the route returns a non-loopback address."""
+    with patch(
+        "homeassistant.components.network.util.socket.socket.getsockname",
+        return_value=[_NO_LOOPBACK_IPADDR],
+    ), patch(
+        "homeassistant.components.network.util.ifaddr.get_adapters",
+        return_value=_generate_mock_adapters(),
+    ):
+        assert await async_setup_component(hass, network.DOMAIN, {network.DOMAIN: {}})
+        await hass.async_block_till_done()
+
+    network_obj = hass.data[network.DOMAIN]
```

Comment:
We could use the helper(s) here and not have to access integration details in the tests. But if there's no use case outside the tests, we don't need to add it.

Reply:
I can't think of a use case currently.